realsense2_camera/src/profile_manager.cpp

```diff
@@ -333,12 +406,69 @@ void VideoProfilesManager::registerVideoSensorParams()
                     }
                     else
                     {
-                        ROS_ERROR_STREAM("Given value, " << parameter.get_value<std::string>() << " is invalid. Set ROS param back to: " << crnt_profile_str.str());
+                        ROS_ERROR_STREAM("Given value, " << parameter.get_value<std::string>() << " is invalid. "
+                                                    << "Run 'ros2 param describe <your_node_name> " << parameter.get_name() 
+                                                    << "' to get the list of supported profiles. "
+                                                    << "Setting ROS param back to: " << crnt_profile_str.str());
                     }
                     _params.getParameters()->queueSetRosValue(parameter.get_name(), crnt_profile_str.str());
                 }
             }, crnt_descriptor);
     _parameters_names.push_back(param_name);
+
+    for (auto sip : sips)
+    {
+        std::string param_name(_module_name + ".profile." + STREAM_NAME(sip) + "_stream_format");
+
+        if (_formats.find(sip) == _formats.end())
+        {
+            _formats[sip] = RS2_FORMAT_ANY;
+        }
+
+        std::string param_value = rs2_format_to_string(_formats[sip]);
+        rcl_interfaces::msg::ParameterDescriptor crnt_descriptor;
+        crnt_descriptor.description = "Available options are:\n" + getProfileFormatsDescriptions(sip);
+        _params.getParameters()->setParam(param_name, param_value, [this, sip](const rclcpp::Parameter& parameter)
+                {
+                    std::string format_str(parameter.get_value<std::string>());
+                    rs2_format temp_format = RS2_FORMAT_ANY;
+                    bool found = false;
+
+                    if (format_str == rs2_format_to_string(RS2_FORMAT_ANY))
+                    {
+                        ROS_WARN_STREAM("A suitable format for " << STREAM_NAME(sip) << " stream will be chosen automatically. "
+                                << "re-enable the stream for the change to take effect.");
+                        found = true;
+                        _formats[sip] = RS2_FORMAT_ANY;
+                    }
+                    else if (string_to_rs2_format(format_str , &temp_format))
```

Comment:
Please change this function name to be something more clear.
I expect it only to convert from string to rs2_format, but it return a Boolean that we need to check.
Or, maybe, change the design of these lines..

Reply:
Have updated the design of this function. Will return RS2_FORMAT_ANY if not found. Should I change the function name as well?